x/profiles/keeper/keeper_app_links.go

```diff
@@ -122,3 +130,20 @@ func (k Keeper) DeleteAllUserApplicationLinks(ctx sdk.Context, user string) {
 		store.Delete(types.UserApplicationLinkKey(link.User, link.Data.Application, link.Data.Username))
 	}
 }
+
+// DeleteExpiredApplicationLinks deletes all the expired application links in the given context
+func (k Keeper) DeleteExpiredApplicationLinks(ctx sdk.Context) {
+	k.IterateExpiringApplicationLinks(ctx, func(_ int64, link types.ApplicationLink) (stop bool) {
+		ctx.EventManager().EmitEvent(
```

Comment:
It would create a lot of index to a block, should we emit only event called `EventTypeApplicationLinksDeleted` outside of iterator instead?

Reply:
🤔 This make sense to me. @RiccardoM any comment about it?